CompilerSource/compiler/components/parse_and_link.cpp

```diff
@@ -245,26 +249,15 @@ int lang_CPP::compile_parseAndLink(const GameData &game, CompileState &state) {
       int ev_id;
         if (event.has_name()) {
           edbg << "  Event[" << event.type() << ", " << event.name() << "] ";
-          user << "FATAL: Unsupported: Event names. Translation required using "
-                  "events.res. Translation is present but there is no logic to "
-                  "use it.";
-
-          // TODO: Using the event type from events.res, translate the name
-          // string to a number and a valid identifier for use in this routine.
-          // It may be wise to store this information in a normalized table
-          // within GameData, similar to how fonts store glyph metrics.
-          // Also, remember to update calls to event_get_human_name.
-
-          // ev_id = translate_event_name(event.name());
-          return E_ERROR_COMPILER_LOGIC;
+          ev_id = object_name_ids[event.name()];
         } else {
           edbg << "  Event[" << event.type() << ", " << event.number() << "] ";
           ev_id = event.number();
         }
 
         //For each individual event (like begin_step) in the main event (Step), parse the code
         parsed_event &pev = pob->events[ev_count++]; //Make sure each sub event knows its main event's event ID.
-        pev.mainId = event.type(), pev.id = event.number();
+        pev.mainId = event.type(), pev.id = ev_id;
```

Comment:
How on earth does this not break every game? The `main_id` and `id` are not both `ev_id`.

Reply:
What? You literally set `ev_id = event.number()` which is the subid! It definitely works.

Reply:
Good then.